Improved the queues.md page to add a little extra info about testing (#1008)
vapor/docs: docs/advanced/queues.md

Queues also has community-based drivers:
- [QueuesMongoDriver](https://github.com/vapor-community/queues-mongo-driver)

> - [QueuesFluentDriver](https://github.com/vapor-community/vapor-queues-fluent-driver)

Reviewer: On second thought, I don't think this should be updated until I've made a non-beta release of the updated version, which hasn't happened yet.

Author: ok! then what is your recomendation? close the P/R? revert this?

- [QueuesDatabaseHooks](https://github.com/vapor-community/queues-database-hooks)
- [QueuesDash](https://github.com/gotranseo/queues-dash)

## Testing

> To avoid syncronization problems and get a deterministic testing, the Queues package provides an XCTQueue library and an AsyncTestQueuesDriver driver dedicated to testing. that you can use as follows:

Reviewer: Grammar and spelling nits:
```suggestion
To avoid synchronization problems and ensure deterministic testing, the Queues package provides an `XCTQueue` library and an `AsyncTestQueuesDriver` driver dedicated to testing which you can use as follows:
```

```swift
final class UserCreationServiceTests: XCTestCase {
    var app: Application!

    override func setUp() async throws {
        self.app = try await Application.make(.testing)
        try await configure(app)

        // Override the driver being used for testing
        app.queues.use(.asyncTest)
    }

    override func tearDown() async throws {
        try await self.app.asyncShutdown()
        self.app = nil
    }
}
```

> See more in the Blog post from [Romain puclet](https://romain.codes/2024/10/08/using-and-testing-vapor-queues/)

Reviewer: Grammar and spelling nits:
```suggestion
See more details in [Romain Pouclet's blog post](https://romain.codes/2024/10/08/using-and-testing-vapor-queues/).
```
